src/lib/shell/directory_stack.rs

```diff
@@ -163,6 +163,7 @@ impl DirectoryStack {
             Action::Push(dir) => {
                 let index = if keep_front { 1 } else { 0 };
                 self.insert_dir(index, dir, variables);
```

Comment:
Could issue #696 be solved if `.` and `..`  get expanded here? In bash that's the default behavior.

Reply:
It could be however I'm not sure that it should. I feel like `.` and `..` should be expanded long before this point, even before the `pushd` builtin receives them.